src/Streamly/FileSystem/Event.hs

```diff
@@ -0,0 +1,109 @@
+-- |
+-- Module      : Streamly.FileSystem.Event
+-- Copyright   : (c) 2020 Composewell Technologies
+--
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+
+module Streamly.FileSystem.Event
+    ( Config
+    , Event
+    , Toggle
+    , defaultConfig
+    , setAllEvents
+
+    -- ** Events of Interest
+    , setRootMoved
+
+    -- ** Watch APIs
+    , watch
+
+    -- * Handling Events
+    , getAbsolutePath
+
+    -- ** Item CRUD events
+    , isCreated
+    , isDeleted
+    , isMoved
+    , isModified
+
+    -- ** Exception Conditions
+    , isOverflow
+
+    -- * Debugging
+    , showEvent
+    )
+where
+
+import Data.List.NonEmpty (NonEmpty)
+import Data.Word (Word8)
+import Streamly.Prelude (SerialT)
+import Streamly.Internal.Data.Array.Storable.Foreign (Array)
+
+#if defined(CABAL_OS_DARWIN)
+import qualified Streamly.Internal.FileSystem.Event.Darwin as Event
+#elif defined(CABAL_OS_LINUX)
+import qualified Streamly.Internal.FileSystem.Event.Linux as Event
+#elif defined(CABAL_OS_WINDOWS)
+import qualified Streamly.Internal.FileSystem.Event.Windows as Event
+#else
+#error "FS Events not supported on this platform
+#endif
+
+-------------------------------------------------------------------------------
+-- Main
+-------------------------------------------------------------------------------
+type Toggle = Event.Toggle
+type Config = Event.Config
+type Event = Event.Event
+
+setRootMoved :: Toggle -> Config -> Config
+setRootMoved = Event.setRootMoved
+
+setAllEvents :: Toggle -> Config -> Config
+#if defined(CABAL_OS_DARWIN)
+setAllEvents = defaultConfig
+#else
+setAllEvents = Event.setAllEvents
+#endif
+
+getAbsolutePath :: Event -> String
+getAbsolutePath = Event.getAbsPath
+
+isCreated :: Event -> Bool
+isCreated = Event.isCreated
+
+isDeleted :: Event -> Bool
+isDeleted = Event.isDeleted
+
+isMoved :: Event -> Bool
+#if defined(CABAL_OS_DARWIN)
+isMoved = Event.isMoved
+#else
+isMoved ev = Event.isMovedFrom ev || Event.isMovedTo ev
```

Comment:
Maybe create an `isMoved` in the modules for Linux and windows with the current definition? Then you can just say `isMoved = Event.isMoved`. You needn't follow it. This looks OK too.

Reply:
keeping as it is.